Orm/Xtensive.Orm/Tuples/TupleDescriptor.cs

```diff
@@ -136,13 +136,13 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
     /// <inheritdoc/>
     public override string ToString()
     {
-      var sb = new StringBuilder();
+      var sb = new ValueStringBuilder(stackalloc char[4096]);
```

Comment:
if we take longest name among supported types it will be ```DateTimeOffset``` (14 chars), Lets round it up to 20 chars plus additional 2 chars between items and it is 20 * FieldCount + 2 * (FieldCount - 1) => 22 * FieldCount - 2. In this case ```char[4096]``` will fit at least 186-field descriptor. If we take 14 chars as longest type name - at least 255-field descriptor. Everything which is smaller will waste a bit of memory These numbers are just to see the perspective.

Do we need to calculate char array length statistically? like 22 * FieldCount or 16 * FieldCount? It will be reasonable on extra-short or extra-long descriptors. But I see array renting inside ```ValueStringBuilder``` and probably renting arrays with length of power of 2 is more reasonable.

Reply:
Renting array is a last resort when stack-allocated buffer is overflowed.
And stack allocation is almost free for non-recursive functions.
Why to bother about this?

As I remember rented arrays have a length== `2^n` no matter how it was requested. If you  call `arrayPool.Rent(100)` it will return array of length `128`, and `ValueStringBuilder` will use all this 128-char buffer